src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/CaseSummaryOverviewServiceTest.kt

```diff
@@ -0,0 +1,137 @@
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Address
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.AddressStatus
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.AllOffenderDetailsResponse
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.ContactDetails
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.OffenderManager
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.ProviderEmployee
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Staff
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.Team
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.communityapi.TrustOfficer
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.exception.PersonNotFoundException
+import java.time.LocalDate
+
+@ExtendWith(MockitoExtension::class)
+@ExperimentalCoroutinesApi
+class CaseSummaryOverviewServiceTest {
+
+  private lateinit var caseSummaryOverviewService: CaseSummaryOverviewService
+
+  @Mock
+  private lateinit var communityApiClient: CommunityApiClient
+
+  @BeforeEach
+  fun setup() {
+    caseSummaryOverviewService = CaseSummaryOverviewService(communityApiClient)
+  }
+
+  @Test
+  fun `throws exception when no person details available`() {
+    runBlockingTest {
+      given(communityApiClient.getAllOffenderDetails(anyString()))
+        .willReturn(Mono.empty())
+
+      val nonExistentCrn = "this person doesn't exist"
+      assertThatThrownBy {
+        runBlockingTest {
+          caseSummaryOverviewService.getOverview(nonExistentCrn)
+        }
+      }.isInstanceOf(PersonNotFoundException::class.java)
+        .hasMessage("No details available for crn: $nonExistentCrn")
+
+      then(communityApiClient).should().getAllOffenderDetails(nonExistentCrn)
+    }
+  }
+
+  @Test
+  fun `retrieves case summary when no offences available`() {
+    runBlockingTest {
+      val crn = "my wonderful crn"
+      given(communityApiClient.getAllOffenderDetails(anyString()))
+        .willReturn(Mono.fromCallable { allOffenderDetailsResponse })
+      given(communityApiClient.getConvictions(anyString()))
+        .willReturn(Mono.empty())
+
+      val response = caseSummaryOverviewService.getOverview(crn)
+
+      val personalDetails = response.personDetails!!
+      val offencesShouldBeEmpty = response.offences
+
+      assertThat(personalDetails.crn).isEqualTo(crn)
+      assertThat(personalDetails.age).isEqualTo(39)
+      assertThat(personalDetails.gender).isEqualTo("Male")
+      assertThat(personalDetails.dateOfBirth).isEqualTo(LocalDate.parse("1982-10-24"))
+      assertThat(personalDetails.name).isEqualTo("John Smith")
+      assertThat(offencesShouldBeEmpty).isNull()
+      then(communityApiClient).should().getAllOffenderDetails(crn)
+    }
+  }
```

Comment:
Think you're missing a test to show the happy path for offences when they get returned.

Reply:
That is covered in the integration tests. Shall I add one here too?